File: messaging/trace_context.go
package messaging

import (
	"context"

	"github.com/nats-io/nats.go"
	"go.opentelemetry.io/otel/propagation"
)

func ExtractTraceContext(ctx context.Context, msg *nats.Msg) context.Context {
	propagator := propagation.TraceContext{}
	return propagator.Extract(ctx, propagation.HeaderCarrier(msg.Header))
}

func InjectTraceContext(ctx context.Context, header nats.Header) nats.Header {
	propagator := propagation.TraceContext{}
	propagator.Inject(ctx, propagation.HeaderCarrier(header))
	return header
}

File: agents/fileprovider/trace_context.py
"""TraceContext propagation helpers for NATS headers."""

from __future__ import annotations

from typing import Dict

from opentelemetry.propagate import inject


def inject_trace_context(headers: Dict[str, str] | None = None) -> Dict[str, str]:
    carrier: Dict[str, str] = headers or {}
    inject(carrier)
    return carrier
